build/test.js

```diff
@@ -36,6 +36,10 @@ function runFrontendUnitTests(singleRun, doneFn) {
   };
 
   let server = new karma.Server(localConfig, function(failCount) {
+    if (failCount > 0) {
+      // Kill backend server and cluster, if running.
+      gulp.start('kill-all');
```

Comment:
Ok, lets make it save first and fancy later. We aim for the following synchronous sequence in CI:

local-up-cluster
check

To do so:
1. let's add wait-for-cluster dependency to local-up-cluster and add doneFN function to make the cluster creation synchronous. 

2. Currently, check:create-cluster has a dependency to local-up-cluster and check. Currently, I assume local-up-cluster and check would still run parallel. A dependency from check to local-up-cluster would solve that, however, I don't want this dependency. So, I guess we can't escape from some 'dirty hack'

e.g. pseudo code:

 check:create-cluster [local-up-cluster] {
   gulp.call(check)
}

or

 check:create-cluster {
     runInSequence(local-up-cluster,check)
}

Does it make sense?